spark3/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java

```diff
@@ -70,21 +82,22 @@ protected BaseProcedure(TableCatalog tableCatalog) {
   }
 
   protected Identifier toIdentifier(String identifierAsString, String argName) {
-    Preconditions.checkArgument(identifierAsString != null && !identifierAsString.isEmpty(),
-        "Cannot handle an empty identifier for argument %s", argName);
-
-    CatalogAndIdentifier catalogAndIdentifier = Spark3Util.catalogAndIdentifier(
-        "identifier for arg " + argName, spark, identifierAsString, tableCatalog);
-
-    CatalogPlugin catalog = catalogAndIdentifier.catalog();
-    Identifier identifier = catalogAndIdentifier.identifier();
+    CatalogAndIdentifier catalogAndIdentifier = toCatalogAdnIdentifier(identifierAsString, argName, tableCatalog);
 
     Preconditions.checkArgument(
-        catalog.equals(tableCatalog),
+        catalogAndIdentifier.catalog().equals(tableCatalog),
         "Cannot run procedure in catalog '%s': '%s' is a table in catalog '%s'",
-        tableCatalog.name(), identifierAsString, catalog.name());
+        tableCatalog.name(), identifierAsString, catalogAndIdentifier.catalog().name());
+
+    return catalogAndIdentifier.identifier();
+  }
+
+  protected CatalogAndIdentifier toCatalogAdnIdentifier(String identifierAsString, String argName,
```

Comment:
I think there is a typo: `toCatalogAdnIdentifier` ->  `toCatalogAndIdentifier`

Reply:
fix!